### DISCUSSION TITLE
Can not get .trivyignore file to work

### DISCUSSION THREAD

**Discussion by christianmaierger** [aquasecurity/trivy#2711] [Q&A] [answered]
Hey there,

my problem is, that the docu is rather vague about how to use the file. On one hand it looks like you just fill in Ids to ignore and can then just use the trivy command without special options:

```
$ cat .trivyignore
# Accept the risk
CVE-2018-14618

# No impact in our settings
CVE-2019-1543

$ trivy image python:3.4-alpine3.9
```

So I use trivy conf instead of image, but that should not be the dealbreaker, the doc also states, that  `--ignorefile string`  is an option for trivy conf, but I also read, that I could also use trivy fs and then specifiy in a different flag if I want to scan for vulnerabilities and/or config. 

But the thing is, no matter what I do, putting .trivyignore on the highest level of my repos filesystem, where one also puts a readme or a gitignore, or in the same folder where the script resides, that executes the trivy command. I can not get it to work, no matter if I use the flag --ignorefile or not. I tried using the relative path to the file here, like ../../.trivyignore, as well as just /.trivyignore, but the Ids I specify are never ignored.

Can someone please tell me what is the correct way to reference the ignore file and where to put it? Thanks so much and sry if the question is kind of a dumb one!

**Comment by christianmaierger** [comment]
Guys, trough trivy with --debug I managed to see, that indeed trivy finds my .trivyignore and even states that the misconfig AVD-KSV-0041 will be ignored just to then find a critical faillure in some files and the link provided points to exactly that: https://avd.aquasec.com/misconfig/rbac/general/avd-ksv-0041/

It seems there is a bug that hinders the ksv-041 from being ignored, or is it possible, I think the docu is rather vague there, that a missconfig can not be ignored by putting it into a trivyignore but hast to by excluded via rego file with a policy? The docu for older versions states it is possible, but for the newest version under misconfig/filtering one just finds the option to filter by severity.

**Reply by knqyf263** [reply]
@afdesk I think detected misconfiguration can be filtered out by .trivyignore. Can you look into it?

**Reply by afdesk** [reply]
@christianmaierger thanks for your report!
I'll try to clarify your question.

**Reply by afdesk** [reply]
@christianmaierger I could reproduce this issue, it looks like there is some problem with IDs.
my test `.yaml` contains  [AVD-KSV-0106](https://avd.aquasec.com/misconfig/kubernetes/general/avd-ksv-0106/)
and i can skip it via new `.trivyignore`:
```
KSV106
```
so I assume that next ignore file can help you:
```
KSV041
```

**Reply by christianmaierger** [reply]
@knqyf263 @afdesk 

Thanks guys!

> @christianmaierger I could reproduce this issue, it looks like there is some problem with IDs. my test `.yaml` contains [AVD-KSV-0106](https://avd.aquasec.com/misconfig/kubernetes/general/avd-ksv-0106/) and i can skip it via new `.trivyignore`:
> 
> ```
> KSV106
> ```
> 
> so I assume that next ignore file can help you:
> 
> ```
> KSV041
> ```

Yes, this is the solution, I tried some variants like KSV-041 etc. but unfortunately not this one until you mentioned it, thanks.

**Reply by afdesk** [reply]
@christianmaierger thanks for your report.
we're adding skipping misconfigurations by AVD ID

**Comment by sanmai-NL** [comment]
Pretty major defect.

There's [nothing to suggest in the docs](https://aquasecurity.github.io/trivy/v0.19.2/vulnerability/examples/filter/) that the standard AVD IDs don't work and which ones do.
Trivy should validate the `.trivyignore` file in the first place.

**Comment by bateau84** [comment]
Did this ever get implemented? i can`t seem to get it work

**Reply by itaysk** [reply]
yes this was implemented

**Reply by chadmyers** [reply]
We're unable to get this to work with AVD AWS ID's despite trying various formats of the ID.  If I want to ignore `AVD-AWS-0104` in my `.trivyignore` file for a `trivy conf` misconfiguration scan, what do I need to put in the trivyignore file?  `AVD-AWS-1014`  doesn't seem to work like I would expect.   Confirmed we are using the `latest` tag of the aquasec/trivy docker image as of today which I believe is 0.38.3

**Reply by AnaisUrlichs** [reply]
Hi @chadmyers I just tried it, not with the specific AVD you shared but generally, screenshots are below

with the trivyignore:
<img width="1443" alt="Screenshot 2023-03-30 at 13 35 25" src="https://user-images.githubusercontent.com/33576047/228838150-f5b6ebe9-608c-4103-ac2d-c89c61f1d027.png">

vs the .trivyignore being commented out
<img width="1453" alt="Screenshot 2023-03-30 at 13 35 42" src="https://user-images.githubusercontent.com/33576047/228838164-949927f5-4572-4a45-bd76-bde81a684210.png">

Are you sure you are on the latest Trivy version?
`trivy version`

**Reply by chadmyers** [reply]
Anais, thank you for the reply!   It turns out it was some confusion on my end about how to reference the .trivyignore when invoking trivy via `docker run` but I got it sorted and it's working now thank you!

**Reply by AnaisUrlichs** [reply]
Great, thank you for letting me know -- we will update the documentation to make it clearer

**Reply by mariusmuntean** [reply]
What is the correct way of specifying the .trivyignore file when running trivy via docker run?

**Reply by chadmyers** [reply]
@mariusmuntean You just need to mount it into your container like you would any other file. The problem we were having is that we had some typos and such. 

Here's an example of our run command (on Windows host OS using Docker Desktop and WSLv2 Linux Containers):

```
& docker run --rm -v "c:\my\local\terraform:/root/terraform" -v "c:\my\local\trivy-results:/root/trivy-results" --rm -v "c:\my\local\.trivyignore:/root/terraform/.trivyignore" -w /root/terraform aquasec/trivy:latest config -f json -o /root/trivy-results/v41/output.json /root/terraform
```

**Reply by mariusmuntean** [reply]
@chadmyers thanks for the example. It helped me to figure it out. 
I suspected that's what's necessary but I had no idea where exactly to mount the file.
It turns out you can just mount it in the working directory (`-w` flag) and name it `.trivyignore`.

**Comment by itaysk** [comment, marked as answer]
fixed in https://github.com/aquasecurity/trivy/pull/2743 and available since v0.31.3
